Foundation/src/DateTimeParser.cpp

```diff
@@ -18,42 +18,140 @@
 #include "Poco/Exception.h"
 #include "Poco/Ascii.h"
 #include "Poco/String.h"
+#include <iostream>
 
+namespace {
+	using parse_iter = std::string::const_iterator;
 
-namespace Poco {
+	[[nodiscard]] parse_iter skip_non_digits(parse_iter it, parse_iter end)
```

Comment:
Shall these helper functions be inlined?

Reply:
No. `inline` is used in headers to avoid ODR issues. Free functions in cpp files should be either static or in an anonymous namespace to avoid IFNDR issues.

Reply:
@andrewauclair please don't introduce new naming conventions, [we have our own](https://github.com/pocoproject/poco/blob/4f1cf683079cc3b00be96259fc42f5c291ab4c77/doc/CppCodingStyleGuide.doc)